### PR TITLE
move youtube atom duration rendering server side

### PR DESCRIPTION
**Pull request by gidsg**
## What does this change?
Moves the formatted duration on the youtube player overlay server side based on the Atom data.
Previously this was done client side using the YouTube iframe API. 

## What is the value of this and can you measure success?
Less DOM mutation, entirety of poster rendered server side.

## Does this affect other platforms - Amp, Apps, etc?
No, AMP uses the standard `amp-youtube` tag.

## Screenshots
N/A 
this is a refactor

## Tested in CODE?
No

### CODE REVIEW

File: common/test/views/support/cleaner/AtomCleanerTest.scala [hunk 1]
```diff
@@ -89,7 +89,7 @@ class AtomCleanerTest extends FlatSpec
   "Youtube template" should "include duration" in {
     val html = views.html.fragments.atoms.media(media = youTubeAtom.map(_.media.head).get, displayCaption = false, mediaWrapper = None)(TestRequest())
     val doc = Jsoup.parse(html.toString())
-    doc.getElementsByClass("youtube-media-atom__bottom-bar__duration").asScala.headOption should be(defined)
+    doc.getElementsByClass("youtube-media-atom__bottom-bar__duration").html() should be("0:36")
   }
```
Comment (TBonnin) [inline common/test/views/support/cleaner/AtomCleanerTest.scala]: Would it make sense to test the `formattedDuration` function with the most common/edge cases?

File: common/app/model/content/Atoms.scala [hunk 2]
```diff
@@ -38,9 +39,30 @@ final case class MediaAtom(
...
+      .appendSuffix(":")
+      .printZeroAlways
+      .minimumPrintedDigits(2)
+      .appendSeconds
+      .toFormatter
+
+    duration.map(d => {
+      val duration =  new Duration(Duration.standardSeconds(d))
+      duration match {
+        case lessThanOneMinute if duration.isShorterThan(new Duration(Duration.standardMinutes(1))) => "0:" + formatter.print(duration.toPeriod())
```
Comment (TBonnin) [inline common/app/model/content/Atoms.scala]: would setting `minimumPrintedDigits(1)` before the minutes take care of this case?

Comment (gidsg) [inline common/app/model/content/Atoms.scala]: @TBonnin No, the `minimumPrintedDigits` is only useful to add padding to some value e.g. `3` becomes `03` but in this case there are no minutes, so it would not print a `0`,

File: common/app/model/content/Atoms.scala [hunk 3]
```diff
@@ -38,11 +38,32 @@ final case class MediaAtom(
...
+    duration.map(d => {
+      val jodaDuration = new Duration(Duration.standardSeconds(d))
+      val durationMillis = jodaDuration.getMillis
+
+      jodaDuration match {
+        case lessThanOneHour if jodaDuration.isShorterThan(new Duration(Duration.standardHours(1))) => stripLeadingZero(DurationFormatUtils.formatDuration(durationMillis, "mm:ss", true))
+        case _ => stripLeadingZero(DurationFormatUtils.formatDuration(durationMillis, "HH:mm:ss", true))
+      }
+    }
+    )
```
Comment (TBonnin) [inline common/app/model/content/Atoms.scala]: would something like this be simpler: 
```
val oneHour = new Duration(Duration.standardHours(1)))
val durationPattern = if(jodaDuration.isShorterThan(oneHour) "mm:ss" else "HH:mm:ss"
val formattedDuration = DurationFormatUtils.formatDuration(durationMillis, pattern, true)
"^0".r.replaceFirstIn(formattedDuration, "") //removing leading zero
```

**Review by TBonnin — APPROVED** [review]
